daemon/containerd/image_list.go

```diff
@@ -91,12 +93,40 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 			return nil, err
 		}
 
+		var repoTags, repoDigests []string
+		rawImg := img.Metadata()
+		target := rawImg.Target.Digest
+
+		logger := logrus.WithFields(logrus.Fields{
+			"name":   img.Name(),
+			"digest": target,
+		})
+
+		ref, err := reference.ParseNamed(rawImg.Name)
+		if err == nil {
+			repoTags = append(repoTags, reference.TagNameOnly(ref).String())
+
+			digested, err := reference.WithDigest(reference.TrimNamed(ref), target)
+			if err != nil {
+				logger.WithError(err).Error("failed to create digested reference")
+			} else {
+				repoDigests = append(repoDigests, digested.String())
+			}
+		}
+
+		if err != nil {
```

Comment:
Should this be an `else` or do we want to go inside in both cases if `reference.ParseNamed(rawImg.Name)` or `reference.WithDigest` fail?

Reply:
Yes, should be an else, thanks. Reordered the branches (to make the `err != nil` first as it's more idiomatic).